Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +123,151 @@ public void Create_PropertyDeclaredInBaseClass ()
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Create_BuildsFuncForPropertyAccess ()
     {
+      var type = new Customer() { UserName = "Test" };
+
+      Assert.That(_addingPropertyValidationRuleCollectorType.PropertyFunc(type), Is.EqualTo("Test"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_InitializesDeferredInitializationValidationMessages ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var propertyValidator = ((IPropertyValidationRule)result).Validators.ToArray()[0] as NotNullValidator;
```

Comment:
Actually I think we might want to use First() here. We also probably don't want to use as, and rather add an extra Assert for which type of validator it is (so no casting in the variable assignment). Then we can safely cast the validator in the already existing assert. I think that might make this test more understandable.

Reply:
Try it, and if it looks good, please also do on the other tests.

Reply:
I think Mike or Patrick once told me to not use `.First()` in the tests and its better to use `.ToArray()[0]`
The cast is actually necessary because otherwise I cannot access ValidationMessage

Reply:
I guess its even better to compare the whole list?

Reply:
Yea, comparing the whole list would be optimal.

Reply:
Since I want to assert the validationMessages, it doesn't make sense to compare the validator instances, because the test will also pass if no validationMessage was initialized. So, I have to do it as I did before